src/main/java/hudson/remoting/jnlp/Main.java

```diff
@@ -217,7 +242,7 @@ public static void _main(String[] args) throws IOException, InterruptedException
         p.parseArgument(args);
         if(m.args.size()!=2)
             throw new CmdLineException("two arguments required, but got "+m.args);
-        if(m.urls.isEmpty())
+        if(m.urls.isEmpty() && !m.disableHttpEndpointCheck)
             throw new CmdLineException("At least one -url option is required.");
 
```

Comment:
if `disableHttpEndpointCheck` is true, it should complain if `instanceIdentity` is not provided. Otherwise it issues the following message (when running the sample command given in the PR description)

```
GRAVE: X-Instance-Identity parameter seems to be invalid
java.io.IOException: X-Instance-Identity parameter seems to be invalid
	at org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver.resolve(JnlpAgentEndpointResolver.java:234)
	at hudson.remoting.Engine.innerRun(Engine.java:574)
	at hudson.remoting.Engine.run(Engine.java:523)
```

which is cryptic if you don't know it has to be provided through command line.

Reply:
see #337